src/Xamarin.Android.Build.Tasks/Linker/MonoDroid.Tuner/MarkJavaObjects.cs

```diff
@@ -196,7 +199,38 @@ void PreserveInvoker (TypeDefinition type)
 			PreserveConstructors (type, invoker);
 			PreserveIntPtrConstructor (invoker);
 			PreserveInterfaceMethods (type, invoker);
+#if ILLINK
+			PreserveInterfaces (invoker);
+#endif // ILLINK
+		}
+
+#if ILLINK
+		void PreserveInterfaces (TypeDefinition type)
+		{
+			if (!type.HasInterfaces)
+				return;
+
+			// Return if [Register(DoNotGenerateAcw=true)] is on the type
+			if (type.HasCustomAttributes) {
+				foreach (var attr in type.CustomAttributes) {
+					if (attr.AttributeType.FullName == "Android.Runtime.Register") {
+						foreach (var property in attr.Properties) {
+							if (property.Name == "DoNotGenerateAcw") {
+								if (property.Argument.Value is bool value && value)
+									return;
+								break;
+							}
+						}
+						break;
+					}
+				}
+			}
```

Comment:
Not sure this code is working, getting roughly the same size `Mono.Android.dll`:
```
+100,284 assemblies/Mono.Android.dll
```
I can investigate more next week.

Reply:
Latest changes this looks better:

![image](https://user-images.githubusercontent.com/840039/188909832-50e04ff8-ef5f-4b27-947d-3753ffe48c7c.png)
![image](https://user-images.githubusercontent.com/840039/188909795-48ff0f0a-4036-4f97-8015-2925c8261f5e.png)